external-crates/move/crates/move-package/src/lib.rs

```diff
@@ -257,37 +252,25 @@ impl BuildConfig {
         // resolution graph diagnostics are only needed for CLI commands so ignore them by passing a
         // vector as the writer
         let resolved_graph = self.resolution_graph_for_package(path, None, writer)?;
-        if resolved_graph.contains_renaming().is_some() {
-            return Err(anyhow!(
-                "Cannot build Move model for package with address renamings"
-            ));
-        }
         let _mutx = PackageLock::lock(); // held until function returns
         model_builder::build(resolved_graph, writer)
     }
 
-    // NOTE: If there are no renamings, then the root package has the global resolution of all named
-    // addresses in the package graph in scope. So we can simply grab all of the source files
-    // across all packages and build the Move model from that.
-    // TODO: In the future we will need a better way to do this to support renaming in packages
-    // where we want to support building a Move model.
     // This deterministically selects unique address names for every named address in the package.
     // It returns the original address assignment to the caller.
+    //
+    // For `derive_address_set` we always derive an address value for any named address if `None`,
+    // otherwise only derive new address values for named addresses with existing values in this
+    // set.
     pub fn move_model_for_package_with_derived_addresses<W: Write>(
         self,
         path: &Path,
+        derive_address_set: Option<&BTreeSet<AccountAddress>>,
```

Comment:
maybe just `&BTreeSet`? though the Option is fine too for clarity. 

I suggest it only because the behavior is the same for `None` and `Some(&BTreeSet::empty())`

Reply:
ah... I kinda took your suggestion higher up after reading and went a bit further (and I think it's actually kinda nicer). Happy to roll it back to this if you want though.

Reply:
It was about removing the block of code that did the address derivation rather than just the checking against zero bit (or at least that's how I read it :p). I went for the "pass a set in with the addresses that you want to derive for" and then realized that passing a function to do this + pulling it up into the CLI layer as opposed to the package layer made things a lot cleaner IMO.